projects/go-lib/src/lib/components/go-portal/go-portal.service.ts

```diff
@@ -0,0 +1,31 @@
+import {Injectable, TemplateRef, ViewContainerRef} from '@angular/core';
+
+@Injectable({providedIn: 'root'})
+export class GoPortalService {
+  private targets: Map<string, ViewContainerRef>;
+
+  constructor() {
+    this.targets = new Map<string, ViewContainerRef>();
+  }
+
+  addTarget(targetName: string, viewContainer: ViewContainerRef): void {
+    this.targets.set(targetName, viewContainer);
+  }
+
+  attachToTarget(targetName: string, template: TemplateRef<any>): void {
+    if (this.targets.has(targetName)) {
+      this.getTarget(targetName).createEmbeddedView(template);
+    }
+  }
+
+  clearTarget(targetName: string): void {
+    if (this.targets.has(targetName)) {
+      this.getTarget(targetName).clear();
+    }
```

Comment:
Same thing here, this can be simplified to:

```
this.getTarget(targetName)?.clear();
```

Reply:
see comment above.